src/SLCore.IntegrationTests/FileAnalysisTests.cs

```diff
@@ -28,41 +28,91 @@
     }
     
     [TestMethod]
-    public async Task Sloop_DefaultRuleConfiguration_SecretsAnalysisProducesExpectedNumberOfIssues()
+    public async Task DefaultRuleConfig_SecretsAnalysisProducesExpectedNumberOfIssues()
     {
-        var relativeFilePath = @"Resources\TwoIssues.js";
-        
-        var issuesByFileUri = await RunFileAnalysis(relativeFilePath);
+        var issuesByFileUri = await RunFileAnalysis(TwoJsIssuesPath);
         
         issuesByFileUri.Should().HaveCount(1);
-        issuesByFileUri[new FileUri(GetFullPath(relativeFilePath))].Should().HaveCount(2);
+        issuesByFileUri[new FileUri(GetFullPath(TwoJsIssuesPath))].Should().HaveCount(2);
     }
-    
-    private async Task<Dictionary<FileUri, List<RaisedIssueDto>>> RunFileAnalysis(string fileToAnalyzeRelativePath)
+
+    [TestMethod]
+    public async Task StandaloneRuleConfig_JavaScriptAnalysisShouldIgnoreOneIssueOfInactiveRule()
+    {
+        var letRuleConfig = CreateInactiveRuleConfigDtoByKey(LetRuleId);
+
+        var issuesByFileUri = await RunFileAnalysis(TwoJsIssuesPath, letRuleConfig);
+
+        issuesByFileUri.Should().HaveCount(1);
+        issuesByFileUri[new FileUri(GetFullPath(TwoJsIssuesPath))].Should().HaveCount(1);
+    }
+
+    [TestMethod]
+    public async Task StandaloneRuleConfig_SecretsAnalysisShouldIgnoreTwoIssuesOfInactiveRule()
+    {
+        var secretsRuleConfig = CreateInactiveRuleConfigDtoByKey(CloudSecretsRuleId);
+
+        var issuesByFileUri = await RunFileAnalysis(ThreeSecretsIssuesPath, secretsRuleConfig);
+
+        issuesByFileUri.Should().HaveCount(1);
+        issuesByFileUri[new FileUri(GetFullPath(ThreeSecretsIssuesPath))].Should().HaveCount(1);
+    }
+
+    [TestMethod]
+    public async Task StandaloneRuleConfig_CtorParamsUnderThreshold_JavaScriptActiveRuleShouldHaveNoIssue()
+    {
+        var ctorParamsRuleConfig = CreateActiveCtorParamRuleConfigDtoByKey(threshold:ActualCtorParams + 1);
+
+        var issuesByFileUri = await RunFileAnalysis(OneIssueRuleWithParamPath, ctorParamsRuleConfig);
+
+        issuesByFileUri.Should().HaveCount(1);
+        issuesByFileUri[new FileUri(GetFullPath(OneIssueRuleWithParamPath))].Should().HaveCount(0);
+    }
+
+    [TestMethod]
+    public async Task StandaloneRuleConfig_CtorParamsAboveThreshold_JavaScriptActiveRuleShouldHaveOneIssue()
+    {
+        var ctorParamsRuleConfig = CreateActiveCtorParamRuleConfigDtoByKey(threshold:ActualCtorParams - 1);
+
+        var issuesByFileUri = await RunFileAnalysis(OneIssueRuleWithParamPath, ctorParamsRuleConfig);
+
+        issuesByFileUri.Should().HaveCount(1);
+        issuesByFileUri[new FileUri(GetFullPath(OneIssueRuleWithParamPath))].Should().HaveCount(1);
+    }
+
+    private async Task<Dictionary<FileUri, List<RaisedIssueDto>>> RunFileAnalysis(string fileToAnalyzeRelativePath, params KeyValuePair<string, StandaloneRuleConfigDto>[] ruleKeyToRuleConfigDto)
```

Comment:
```suggestion
    private async Task<Dictionary<FileUri, List<RaisedIssueDto>>> RunFileAnalysis(string fileToAnalyzeRelativePath, Dictionary<string, StandaloneRuleConfigDto> ruleConfig = null)
```

This would be better IMO

Reply:
Done